options/options_test.cc

```diff
@@ -145,8 +145,16 @@ TEST_F(OptionsTest, GetOptionsFromMapTest) {
 
```

Comment:
Since we still keep the original interface of GetXXXFromMap, it would be better that we still keep the original tests and we add the new tests for the new interfaces.

Reply:
I would rather deprecate the old interface and have nothing use it.  However, since the old interface is in the public header file and hence public API, I did not think I could delete it entirely.

Would it be acceptable to leave the tests using the new interface and add a few tests that test  the old interface?

Reply:
@mrambacher I had a discussion with other team members in term of this. Yes, the old interfaces are still in use in some products, we need to keep them for some time to ensure backward compatible. Maybe we can gradually remove them in the future. But for now, it would be better to keep the tests for the old interfaces and just add the new tests for the new interfaces.

Reply:
I added tests that explicitly test the old interfaces.